server_api.py

```diff
@@ -928,11 +928,11 @@ def upload_text_insert_db(json_obj):
             invisible_title = text_id
         
         #get file place
-        sql = ("SELECT type_dir FROM data_type WHERE type_id=" + str(type_id))
-        pure_result = db.query(sql)
+        with DataTypeDao() as dataTypeDao:
+            type_dir = dataTypeDao.getTypeDir(typeId=str(type_id))
         try:
             text_system_name = text_id + ".txt"
-            system_file_dir = os.path.join(server_dir, "static/"+str(pure_result[0][0]))
+            system_file_dir = os.path.join(server_dir, "static/"+str(type_dir))
```

Comment:
Change to `os.path.join(server_dir,"static",str(type_dir))`, or you can check whether the type of `type_dir` is str(), if yes then remove casting to str().